hangman_lunch2.rb

```diff
@@ -0,0 +1,206 @@
+  def draw_board
+    if @wrong_count == 0
+      puts """
+    ___________________          _-_
+    \\==============_=_/ ____.---'---`---.____
+                \\_ \\    \\----._________.----/
+                  \\ \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+"""
+    elsif @wrong_count == 1
+      puts """
+                                 _-_
+                        ____.---'---`---.____
+                \\_ \\    \\----._________.----/
+                  \\ \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+""".cyan
+    elsif @wrong_count == 2
+      puts """
+                                 _-_
+                        ____.---'---`---.____
+                   \\    \\----._________.----/
+                    \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+      """.blue
+    elsif @wrong_count == 3
+      puts """
+                                _-_
+                       ____.---'---`---.____
+                       \\----._________.----/
+                        /  /   `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+      """.yellow
+    elsif @wrong_count == 4
+      puts """
+                                   _-_
+                          ____.---'---`---.____
+                          \\----._________.----/
+                          /  /    `-_-'
+
+
+
+
+
+      """.magenta
+    elsif @wrong_count == 5
+      puts """
+                                   _-_
+                          ____.---'---`---.____
+                          \\----._________.----/
+                                  `-_-'
+
+
+
+
+
+      """.red
+    else
+      lose
+    end
+
+  end
+
+# Creates blank answer line
+  def answer_line
+      @answer_line = []
+      (@answer.length).times do |ans|
+      @answer_line.push(" _ ")
+    end
+    return @answer_line
+  end
+
+
+# ------------------
+
+# Enter and check answer
+  def play
+    # Check until answer is complete (guesses have revealed all letters)
+    until @answer_line.join == @answer
+      # Get a guess
+      puts "Guess a letter, please:"
+      input = gets.chomp.upcase
+
+      # Allow user to quit
+      if input == "QUIT"
+        exit
+      end
+
+      # Error if invalid input given.
+      until input.length == 1 && @letter_array.include?(input.upcase)
+        puts "You're better than this. Please enter ONE LETTER, thanks."
+        input = gets.chomp.upcase
+      end
+
+      # display guesses thus far
+      if input.length == 1 && @letter_array.include?(input.upcase)
+        @letters_guessed.push(input)
+        puts "Letters Guessed: #{@letters_guessed.join(" ")}"
+      end
+
+      # split answer word into array consisting of each letter.
+      letters = @answer.split("")
```

Comment:
Since you're in a loop, you'd end up doing the split every time you loop. You could define letters before the loop, so it'd only happen once. (outside of the `until`)